NDTensors/src/lib/UnallocatedArrays/test/runtests.jl

```diff
@@ -0,0 +1,207 @@
+@eval module $(gensym())
+using FillArrays: FillArrays, AbstractFill, Fill, Zeros
+using NDTensors: NDTensors
+using NDTensors.UnallocatedArrays
+using LinearAlgebra: norm
+using Test: @test, @testset, @test_broken
+
+include(joinpath(pkgdir(NDTensors), "test", "NDTensorsTestUtils", "NDTensorsTestUtils.jl"))
+using .NDTensorsTestUtils: devices_list
+
+@testset "Testing UnallocatedArrays" for dev in devices_list(ARGS),
+  elt in (Float64, Float32, ComplexF64, ComplexF32)
+
+  @testset "Basic funcitonality" begin
+    z = Zeros{elt}((2, 3))
+    Z = UnallocatedZeros(z, dev(Matrix{eltype(z)}))
+
+    @test Z isa AbstractFill
+    @test size(Z) == (2, 3)
+    @test length(Z) == 6
+    @test sum(Z) == 0
+    @test norm(Z) == 0
+    @test Z[2, 3] == 0
+    @test allocate(Z) isa dev(Matrix{elt})
+    Zp = set_alloctype(z, dev(Matrix{elt}))
+    @test Zp == Z
+    Zc = copy(Z)
+    @test Zc == Z
+    Zc = complex(Z)
+    @test eltype(Zc) == complex(eltype(z))
+    @test Zc[1, 2] == 0.0 + 0.0im
```

Comment:
```suggestion
    @test Zc[1, 2] == zero(elt)
```

Reply:
Or `@test iszero(Zc[1, 2])`.

Reply:
I converted all of the instances which check for zero to use the `iszero` function